python/pyspark/sql/types.py

```diff
@@ -457,13 +466,10 @@ def __init__(self, fields=None):
         """
         if not fields:
             self.fields = []
-            self.names = []
```

Comment:
This is needed for performance (for serialization,  toInternal and fromInternal)